Sources/Danger/BitBucketCloud.swift

```diff
@@ -0,0 +1,228 @@
+
+    /// The PR's source, The repo Danger is running on
+    public let source: BitBucketCloudMergeRef
+
+    /// The pull request's current status.
+    public let state: State
+
+    /// Title of the pull request
+    public let title: String
+
+    /// Date of last update
+    public let updatedOn: Date
+}
+
+public struct BitBucketCloudMergeRef: Decodable {
+    private struct Branch: Decodable, Equatable {
+        let name: String
+    }
+
+    private struct Commit: Decodable, Equatable {
+        let hash: String
+    }
+
+    private let branch: Branch
+
+    public var branchName: String {
+        return branch.name
+    }
+
+    private let commit: Commit
+
+    /// Hash of the last commit
+    public var commitHash: String {
+        return commit.hash
+    }
+
+    public let repository: BitBucketCloudRepo
+}
+
+public struct BitBucketCloudRepo: Decodable, Equatable {
+    private enum CodingKeys: String, CodingKey {
+        case fullName = "full_name"
+        case name
+        case uuid
+    }
+
+    public let fullName: String
+
+    public let name: String
+
+    /// The uuid of the repository
+    public let uuid: String
+}
+
+public struct BitBucketCloudPRParticipant: Decodable, Equatable {
+    public enum Role: String, Decodable {
+        case reviewer = "REVIEWER"
+        case participant = "PARTICIPANT"
+    }
+
+    /// Did they approve of the PR?
+    public let approved: Bool
+
+    /// How did they contribute
+    public let role: Role
+
+    /// The user who participated in this PR
+    public let user: BitBucketCloudUser
+}
+
+public struct BitBucketCloudUser: Decodable, Equatable {
+    private enum CodingKeys: String, CodingKey {
+        case accountId = "account_id"
+        case displayName = "display_name"
+        case nickname
+        case uuid
+    }
+
+    /// The acount id of the user
+    public let accountId: String
+
+    /// The display name of user
+    public let displayName: String
+
+    /// The nick name of the user
+    public let nickname: String
+
+    /// The uuid of the user
+    public let uuid: String
+}
+
+public struct BitBucketCloudCommit: Decodable, Equatable {
+    public struct Author: Decodable, Equatable {
+        /// Format: `Foo Bar <foo@bar.com>`
+        let raw: String
+
+        /// The user that created the commit
+        let user: BitBucketCloudUser
+    }
+
+    public struct Parent {
+        /// The full SHA
+        let hash: String
+    }
+
+    /// The author of the commit, assumed to be the person who wrote the code.
+    public let author: Author
+
+    /// When the commit was commited to the project
+    public let date: Date
+
+    /// The SHA for the commit
+    public let hash: String
+
+    /// The commit's message
+    public let message: String
+}
+
+public struct BitBucketCloudComment: Decodable, Equatable {
+    public struct Inline: Decodable, Equatable {
+        public let from: Int
+        public let to: Int
```

Comment:
<!--
  0 failure: 
  1 warning:  Variable name sho...
  
  
  DangerID: danger-id-Danger;
  File: Sources/Danger/BitBucketCloud.swift;
  Line: 183;
-->

- :warning: Variable name should be between 3 and 40 characters long: 'to' (`identifier_name`)